WIP: Pinephone patches 3

Mindavi opened the pull request:
Opening this PR so @piegamesde can track it.

Can be closed whenever all changes are upstreamed.

Mindavi committed:
colord: disable argyllcms when cross-compiling

It appears the build system for argyllcms is not made for cross-compiling,
so let's disable it for now.

colord has more dependencies that don't work yet, so it won't cross-compile properly yet.

Mindavi committed:
geoclue2: support cross-compilation

Mindavi committed:
lightdm: fix cross-compilation

Mindavi committed:
avahi: support prefixed (native) pkg-config

Artturin commented on pkgs/development/libraries/libgtop/default.nix:
```diff
@@ -26,12 +27,23 @@ stdenv.mkDerivation rec {
   ];
 
+  postPatch = ''
+    substituteInPlace src/daemon/Makefile.am \
+      --replace "\`uname -s\`" "DUMMY_KERNEL_NAME" \
+      --replace "\`uname -r\`" "DUMMY_KERNEL_RELEASE" \
+      --replace "\`uname -v\`" "DUMMY_KERNEL_VERSION" \
+      --replace "\`uname -m\`" "DUMMY_KERNEL_MACHINE"
```
i added deterministic-uname in https://github.com/NixOS/nixpkgs/pull/210102 so this can be removed

uninsane committed:
webkitgtk: support cross compilation

which previously errored near the end of compilation with:
```
/nix/store/nnnn-wayland-aarch64-unknown-linux-gnu-1.22.0-bin/bin/wayland-scanner: line 0: syntax error: unexpected word (expecting ")")
```

that is, the build was trying to invoke the host `wayland-scanner`: both
host and build wayland-scanner are made available to the build
environment and the build script isn't smart enough to choose the right
one. fix by explicitly instructing which `wayland-scanner` to use.

(cherry picked from commit 72b3e9b5a0d838ff3a374f9e35e9da0880c01b85)

abathur committed:
resholve: 0.9.1 -> 0.10.2, ongoing cross fixes

(cherry picked from commit 56d5b181c9f2bf748a44526f4db9861524e1dbb2)

uninsane committed:
gnupg: fix cross compilation

when `npth` upgraded 1.6 -> 1.7, the `npth-config` binary it previously
provided was removed, and functionality folded into `gpgrt-config`. see:
<https://github.com/gpg/npth/blob/417abd56fd7bf45cd4948414050615cb1ad59134/NEWS#L8>

(cherry picked from commit d599839060400762a67d2c01d15b102ffe75e703)

abathur committed:
xdg-utils: override lore for perl

The OSH update (and associated refactoring) in resholve seem
to have uncovered surfaced a fresh error here.

(cherry picked from commit 6cf9a8930a0f16ab1e0f5e88d3b1e1ab4baaeeef)

Mindavi committed:
appstream: fix cross

Mindavi committed:
wlogout: fix cross-compilation to aarch64

Mindavi committed:
pixz: strictDeps (WIP)

Probably some more buildInputs should be moved, but this seems to fix the build at least

Mindavi committed:
wlroots_0_17: fix cross-compilation
